### PR TITLE
Rails capstone

### PR DESCRIPTION
**Pull request by mnaibei**
1. Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

 2. Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).

The user can log into the app using email and password.

If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

 3. Home page (categories page)

When the user logs in, they are presented with the categories page.

For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

 4. Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.

At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).

There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

When the user clicks on the "Back" button (<), the user navigates to the home page.

 5. "Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).

The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.

When the user clicks on the "Back" button (<), the user navigates to the home page.

 6. "Add a new transaction" page

The user fills out a form to create a new transaction with:

name (mandatory)

amount (mandatory)

categories (mandatory at least one)

The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.

video: https://www.loom.com/share/d863045a7763498fa5f9a08580f35748?sid=6c0e5e95-37a5-4227-a899-27eee970211f

### CODE REVIEW

File: app/controllers/records_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @group = Group.find(params[:group_id])
+    @record = @group.records.build
+  end
+
+  def create
+    @group = Group.find(params[:group_id])
+    @record = @group.records.build(record_params)
+    @record.author = current_user
+
+    Rails.logger.info @record.inspect
```
Comment (ichala) [inline app/controllers/records_controller.rb:20]: - [ ] Kindly make sure that all groups and transactions are related only the user that created them , currently all users share and manage the same groups and taxes, if i create an account and then create a group , i can see & manage that same group from another account, kindly fix that for transactions & groups by using current_user provided by devise 🙏

File: app/views/records/index.html.erb [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+  <% if @records.empty? %>
     <div class="no_content">
       <p>No transactions available yet!</p>
     </div>
   <% end %>
+</div>
+
+<div class="group_btn">
+    <%= link_to 'Add New Transaction', new_group_record_path(@group), class: 'add-btn' %>
 </div>
```
Comment (ichala) [inline app/views/records/index.html.erb]: - [ ] as part of the requirements kindly make sure to follow exactly these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:

- Colors
- Typography: font face, size and weight.
- Layout: composition and space between elements.

Your Design           | Required Design
:-------------------------:|:-------------------------:
image | image


Your Design           | Required Design
:-------------------------:|:-------------------------:
image | image

Your Design           | Required Design
:-------------------------:|:-------------------------:
image | image

Your Design           | Required Design
:-------------------------:|:-------------------------:
image | image

File: app/views/records/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  </div>
+
+  <div class="field">
+    <%= form.label :group_ids, class: "form-label" %>
+    <%= form.collection_select :group_ids, Group.all, :id, :name, {}, class: "form-select" %>
+  </div>
+  </div>
+
+  <%= form.submit "Save", class: "submit-btn" %>
+<% end %>
```
Comment (ichala) [inline app/views/records/new.html.erb]: - [ ] as part of the requirements kindly add some validations errors message and some success/fail paths where the user can see the error so he can fix it  , kindly fix it for all of your inputs & forms 🙏 


sasssssssssssssssssssssssss

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- Created Unit & Integration Tests✔️ 
- Good database schema✔️
- Good presentation video ✔️
- 


 ## Required Changes ♻️

- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @mnaibei, and @ichala👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Highly organized code. ✔️
- Nice UI design. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**